File: Easy/Horse-Racing Duals/Horse-Racing Duals.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

class Solution
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());
        int result = int.MaxValue;
        List<int> lst=new List<int>(); 

        for (int i = 0; i < n; i++)
        lst.Add(int.Parse(Console.ReadLine()));
        
        lst.Sort();
        
        for (int i = 1; i < n; i++)
        if (Math.Abs(lst[i]-lst[i-1])<result) result = Math.Abs(lst[i]-lst[i-1]);

        Console.WriteLine(result);
    }
}

File: Easy/Horse-Racing Duals/Horse-Racing Duals.py
lst = []
result = 9999
n = int(input())
for i in range(n):
    lst.append(int(input()))

lst = sorted(lst)

for i in range(n-1):
    if abs(lst[i+1]-lst[i])<result:
        result = abs(lst[i+1]-lst[i])

print(result)